contracts/NonFungibleToken.cdc

```diff
@@ -93,6 +93,24 @@ pub contract interface NonFungibleToken {
         pub fun resolveView(_ view: Type): AnyStruct? {
             return nil
         }
+
+        /// Returns an array of the IDs that are in the collection
+        ///
+        /// @return An array containing all the child IDs on the nft
+        ///
+        pub fun getChildren(): [UInt64] {
+            return []
+        }
+
+        /// Returns a borrowed reference to an child NFT of this NFT
+        /// so that the caller can read data and call methods from it
+        ///
+        /// @param id: The ID of the NFT that want to be borrowed
+        /// @return A reference to the NFT
+        ///
+        pub fun borrowChildNFT(id: UInt64): &NFT? {
+            return nil
+        }
```

Comment:
I'm a little confused about these. First, don't we want to also have the `borrowSubCollection` method here?
Second, what is a child NFT in this context?

Reply:
We cannot borrowSubCollection as &{ResolverCollection}here since it would be an import loop. Also NFT cannot implement MetadataViews.Resolver because of the same import loop.

We can always change the child term to use something else, `subNFT` or embedded or another word.

Reply:
The decition to put NFTCollectionData in MetadataViews and not in here is the cause of this :/

Reply:
I actually think that having subCollection in &{Resolver}and these methods here is confusing, should we not use `borrowSubCollection` in &{Resolver}and standardize on names here?

Reply:
Can't we still have `borrowSubCollection` here and in metadata views and just don't import metadata views like we do with the other resolver methods? Maybe I'm misunderstanding something

Reply:
What is borrowSubCollection here going to return then?

Reply:
One problem here is that there is no interface that covers the case of read access to all sub nfts. CollectionPublic also has deposit.

Reply:
It could just return an optional